src/ansys/sound/core/sound_composer/source_control_spectrum.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Sound composer's spectrum source control."""
+from .._pyansys_sound import PyAnsysSoundException
+from ._source_control_parent import SourceControlParent
+
+METHODS = ("Inverse FFT", "Hybrid")
```

Comment:
Okay, my comment is maybe not appropriate : this variable is only used for printing the method names right ?

Reply:
@a-bouth In the scope of what I did, yes. But you seem to suggest here that there will be other needs for it outside of this class. In which case, this should probably be defined in sound_composer_parent...
To what I understand, you merely considered an int type because operators can't take str inputs. Then shouldn't we store strs, which are more explicit to the user, and only convert it to int where and when the operator is called? (while the correspondence can be defined and centralized in a relevant place as a constant enum or dict, eg in the parent module).
All in all, maybe this would deserve some additional thinking design-wise, don't you think?

Reply:
@ansaminard , operators can take string input but it's best practice to use enums instead of strings (What if we change the spelling of the string ? What if the user uses upper case or lower case etc. )

So, no, I don't think we should use string and I'd rather use a documented `enum`

I do agree with you that it should be in SoundComposerParent so all the child classes can have the same reference

My suggestion would be to have, in `sound_composer_parent.py`

The following lines : 

`
from enum import Enum
class SpectrumSynthesisMethods(Enum):
    INVERSE_FFT = 1 # Add doc here
    HYBRID = 2 # Add doc here
`

Then, in `source_control_spectrum.py` you should be able to import this class easily and use the enums.

That would be my prefered solution

(If you need we can call quickly to discuss that)

Reply:
Just an idea: can't we define the Enum as a class attribute instead? then we wouldn't have to import anything, the child class would inherit from it (and we don't define a property for it, such that it remains constant)

Reply:
Also you will have to provide the doc to put in your code lines. I don't know what these are myself